### PR TITLE
tektoncd-cli: 0.16.0 -> 0.17.0

### PR DESCRIPTION
**Pull request by 06kellyjac**
###### Motivation for this change

Bump tektoncd-cli to `0.17.0`

Also:
- Set buildFlagsArray using bash
- Add installCheckPhase

###### Things done

- [X] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [X] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [x] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [X] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [X] Ensured that relevant documentation is up to date
- [X] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

**Comment by 06kellyjac** [comment]
Result of `nixpkgs-review pr 116036` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>tektoncd-cli</li>
  </ul>
</details>

**Review by vdemeester — APPROVED** [review]
LGTM 🐯

**Review by mstrangfeld — APPROVED** [review]
LGTM :+1:

File: pkgs/applications/networking/cluster/tektoncd-cli/default.nix [hunk 1]
```diff
@@ -40,6 +37,15 @@ buildGoModule rec {
...
   '';
 
+  doInstallCheck = true;
+  installCheckPhase = ''
+    runHook preInstallCheck
+    $out/bin/tkn --help
+    $out/bin/tkn version | grep "Client version: ${version}"
+    runHook postInstallCheck
+  '';
+
```
Comment (SuperSandro2000) [inline pkgs/applications/networking/cluster/tektoncd-cli/default.nix:47]:
```suggestion
  '';
```

File: pkgs/applications/networking/cluster/tektoncd-cli/default.nix [hunk 2]
```diff
@@ -40,6 +37,15 @@ buildGoModule rec {
...
+
   meta = with lib; {
```
Comment (SuperSandro2000) [inline pkgs/applications/networking/cluster/tektoncd-cli/default.nix:49]:
```suggestion
  '';

  meta = with lib; {
```

**Pull request merged**